lightning/src/ln/onion_payment.rs

```diff
@@ -273,7 +273,37 @@ pub(super) fn create_recv_pending_htlc_info(
 			 intro_node_blinding_point.is_none(), true, invoice_request)
 		}
 		#[cfg(trampoline)]
-		onion_utils::Hop::TrampolineReceive { .. } | onion_utils::Hop::TrampolineBlindedReceive { .. } => todo!(),
+		onion_utils::Hop::TrampolineReceive {
+			trampoline_hop_data: msgs::InboundOnionReceivePayload {
+				payment_data, keysend_preimage, custom_tlvs, sender_intended_htlc_amt_msat,
+				cltv_expiry_height, payment_metadata, ..
+			}, ..
+		} =>
+			(payment_data, keysend_preimage, custom_tlvs, sender_intended_htlc_amt_msat,
+				cltv_expiry_height, payment_metadata, None, false, keysend_preimage.is_none(), None),
+		#[cfg(trampoline)]
+		onion_utils::Hop::TrampolineBlindedReceive {
+			trampoline_hop_data: msgs::InboundOnionBlindedReceivePayload {
+				sender_intended_htlc_amt_msat, total_msat, cltv_expiry_height, payment_secret,
+				intro_node_blinding_point, payment_constraints, payment_context, keysend_preimage,
+				custom_tlvs, invoice_request
+			}, ..
+		} => {
```

Comment:
It seems that if I revert the changes to `onion_payment.rs`, the newly added test still passes?